src/util/PathUtil.v3

```diff
@@ -0,0 +1,15 @@
+component PathUtil {
```

Comment:
This is now a little overfit to the extension; it trims off the last 5 characters. To generalize this it should probably scan forward for the last slash and then scan backward for the last '.'. We can also do that without making a string buffer, just keeping track of the start and end and using either `Arrays.range()` or `Ranges.dup(path[start ... end])`.